### PR TITLE
glfw3 3.2

### PR DESCRIPTION
**Pull request by rakkesh**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----

### CODE REVIEW

**Comment by rakkesh** [comment]
```sh
$ brew audit --strict --online glfw3
Warning: Passing arbitrary symbols to `option` is deprecated: :static
Warning: Passing arbitrary symbols to `option` is deprecated: :static
Warning: Passing arbitrary symbols to `option` is deprecated: :static
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-examples"
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-examples"
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-examples"
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-tests"
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-tests"
Warning: Passing arbitrary symbols to `option` is deprecated: :"build-tests"
```

File: glfw3.rb [hunk 1]
```diff
@@ -14,14 +14,10 @@ class Glfw3 < Formula
   option :universal
   option "without-shared-library", "Build static library only (defaults to building dylib only)"
   option "with-examples", "Build examples"
-  option "with-tests", "Build test programs"
+  option "with-test", "Build test programs"
```
Comment (MikeMcQuaid) [inline glfw3.rb]: Need a `deprecated_option` for this.

File: glfw3.rb [hunk 2]
```diff
@@ -14,14 +14,10 @@ class Glfw3 < Formula
...
 
   depends_on "cmake" => :build
 
-  deprecated_option "build-examples" => "with-examples"
-  deprecated_option "static" => "without-shared-library"
-  deprecated_option "build-tests" => "with-tests"
```
Comment (MikeMcQuaid) [inline glfw3.rb]: This need left as-is.

File: glfw3.rb [hunk 3]
```diff
@@ -14,14 +14,10 @@ class Glfw3 < Formula
...
-
   # make library name consistent
   patch :DATA
```
Comment (MikeMcQuaid) [inline glfw3.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

Comment (MikeMcQuaid) [inline glfw3.rb]: Any chance of making it an `inreplace` or similar?

**Comment by rakkesh** [comment]
Closing in favour of https://github.com/Homebrew/homebrew-versions/pull/1305.

**Pull request closed without merging**